File: 6-Python/-Projetos Em Python/7- diagonal_negativos.py
n = int(input("Qual a ordem da matriz? "))
m = n
print()

mat: [[float]] = [[0 for x in range(n)] for x in range(m)]

for i in range(0, n):
    for j in range(0, m):
        mat[i][j] = float(input(f"Elemento [{i},{j}]: "))

print()
print("Diagonal Principal:")
cont = -1
for i in range(0, n):
    cont = cont + 1
    for j in range(0, m):
        if j == cont:
            print(f"{mat[i][j]:.2f} ", end='')

qntd_negat = 0
for i in range(0, n):
    for j in range(0, m):
        if mat[i][j] < 0:
            qntd_negat = qntd_negat + 1

print()
print(f"Quantidade de negativos: {qntd_negat}")

File: 4-Java/-Projetos Em Java/_7_diagonal_negativos.java
import java.util.Scanner;
import java.util.Locale;

public class _7_diagonal_negativos {

	public static void main(String[] args) {
		
		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);
		
		System.out.print("Digite a ordem da matriz: ");
		int n = sc.nextInt();
		int m = n;
		
		double[][] mat = new double[n][m];
		
		System.out.println();
		for (int i = 0; i < n; i++) {
			for (int j = 0; j < m; j++) {
				System.out.print("Elemento [" + i + "," + j + "]: ");
				mat[i][j] = sc.nextDouble();
			}
		}
		
		int contador = 0;
		System.out.println("Diagonal da Matriz:");
		for (int i = 0; i < n; i++) {
			for (int j = 0; j < contador; j++) {
				System.out.println(mat[i][j] + " ");
			}
			contador = contador + 1;
		}
		System.out.println();
		
		int contador_negativos = 0;
		System.out.print("Quantidade de numeros impares: ");
		for (int i = 0; i < n; i++) {
			for (int j = 0; j < m; j++) {
				if (mat[i][j] < 0) {
					contador_negativos = contador_negativos + 1;
				}
			}
		}
		System.out.println(contador_negativos);
		
		sc.close();
	}

}
